Formula/lgogdownloader.rb

```diff
@@ -0,0 +1,32 @@
+class Lgogdownloader < Formula
+  desc "Open source downloader for GOG.com games"
+  homepage "https://sites.google.com/site/gogdownloader"
+  url "https://sites.google.com/site/gogdownloader/lgogdownloader-3.2.tar.gz"
+  sha256 "9e60af50c18533884ca317778b291152d5f04d93dfd3dc2579f6f965538b2031"
+
+  depends_on "boost"
+  depends_on "htmlcxx"
+  depends_on "jsoncpp"
+  depends_on "liboauth"
+  depends_on "rhash"
+  depends_on "tinyxml2"
+
+  depends_on "cmake" => :build
+  depends_on "help2man" => :build
+  depends_on "pkg-config" => :build
+
+  def install
+    mkdir "build" do
+      system "cmake", "..",
+        "-DCMAKE_INSTALL_PREFIX=#{prefix}",
+        "-DCMAKE_BUILD_TYPE=Release"
+
+      system "make", "install"
+    end
+  end
+
+  test do
+    assert_equal "LGOGDownloader #{version}",
+      pipe_output("#{bin}/lgogdownloader --version").chomp
```

Comment:
We need a test that does a bit more… can we try to login and check that it fails?

Reply:
I’ll look into it.

Reply:
The GOG credentials are kept somewhere in the `$XDG_CONFIG_HOME` hierarchy. Those credentials may be correct, incorrect, or missing. How would the test even know?

Reply:
The test will most often run on CI machines, where there are no credentials. So let's assume that it fails to get correct credentials, and check for that.

Reply:
This is what the manpage says about `brew test`:

> `brew test formula` […] should generally indicate to the user if something is wrong with the installed formula.

What value would a login test add to warrant breaking the user contract and very likely confusing users?

Reply:
From the [formula cookbook](https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md):

> We want tests that don't require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.

Printing version number doesn't exercise much of the tool's installation. We want tests that can be run on CI, and tell us whether a formula got broken — for example missing linkage or else — when updated.

Reply:
Thanks for pointing that out @fxcoudert. I agree with you that having more sophisticated tests would be a good thing.

At the same time, user experience is just as important as that. (Imagine a user has an issue with `lgogdownloader` at login time, then runs `brew test` in an attempt to narrow down the problem. If the program itself is fine, then `brew test` should pass. A test failure would mislead the user into thinking there’s a problem with the formula.)

How about we divert `$XDG_CONFIG_HOME` to something like this:

> `FileUtils.mktemp(retain: true) { pwd }`

for the duration of the test? I feel that this might work reliably for both CI and users.

Reply:
I agree. Test itself is run in a temporary directory, so redirecting `XDG_CONFIG_HOME` to `pwd` would work fine.

Reply:
Done.

It was a bit of a pain because it required a TTY.
That said, the test should now be as robust as it gets.